config/tests/src/test/java/org/jboss/windup/config/CategoriesTest.java

```diff
@@ -0,0 +1,282 @@
+    {
+        @Override
+        public Collection<String> getCategories()
+        {
+            return Collections.EMPTY_LIST;
+        }
+    }
+
+    public static class TestCategoryEmptyRuleProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public Collection<String> getCategories()
+        {
+            return Collections.EMPTY_LIST;
+        }
+    }
+
+    public static class TestCategory1RuleProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public Collection<String> getCategories()
+        {
+            return Arrays.asList("Foo");
+        }
+    }
+
+    public static class TestCategory2RuleProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public Collection<String> getCategories()
+        {
+            return Arrays.asList("Foo", "Bar");
+        }
+    }
+
+    public static class TestCategoryEnhanceStringRuleProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public void enhanceMetadata(Context context)
+        {
+            super.enhanceMetadata(context);
+            context.put(RuleMetadata.CATEGORY, "Foo");
+        }
+    }
+
+    public static class TestCategoryEnhanceJoinedStringRuleProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public void enhanceMetadata(Context context)
+        {
+            super.enhanceMetadata(context);
+            context.put(RuleMetadata.CATEGORY, "Foo, Bar");
+        }
+    }
+
+    public static class TestCategoryEnhanceCollectionRuleProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public void enhanceMetadata(Context context)
+        {
+            super.enhanceMetadata(context);
+            context.put(RuleMetadata.CATEGORY, Arrays.asList("Foo", "Bar"));
+        }
+    }
+
+    public static class TestCategoryEnhanceEmptyProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public void enhanceMetadata(Context context)
+        {
+            super.enhanceMetadata(context);
+            context.put(RuleMetadata.CATEGORY, "");
+        }
+    }
+
+    public static class TestCategoryEnhanceUnsetProvider extends TestCategoryRuleProviderBase
+    {
+        @Override
+        public void enhanceMetadata(Context context)
+        {
+        }
+    }
+
+
+
+    // A base for those above.
+    public static class TestCategoryRuleProviderBase extends WindupRuleProvider
+    {
+        @Override
+        public Configuration getConfiguration(GraphContext context)
+        {
+            return ConfigurationBuilder.begin()
+            .addRule(new Rule(){
+
+                @Override
+                public String getId()
+                {
+                    return this.getClass().getSimpleName();
+                }
+
+                @Override
+                public boolean evaluate(Rewrite rwrt, EvaluationContext ec)
```

Comment:
Please rename variables to use full words.

ec -> context or evaluationContext